src/plugins/WebOfTrust/util/BackgroundJobFactory.java

```diff
@@ -0,0 +1,50 @@
+package plugins.WebOfTrust.util;
+
+/**
+ * Factory for {@link BackgroundJob}s which allows all background jobs constructed using an instance
+ * of this class to be terminated with a single method invocation.
+ *
+ * @author bertm
+ * @see BackgroundJobFactoryBase
+ */
+public interface BackgroundJobFactory {
+    /**
+     * Constructs a new background job using the default parameters. When this background job is
+     * {@link BackgroundJob#terminate() terminated}, the running job will be notified by
+     * interruption of its thread. Hence, the job implementer must take care not to swallow
+     * {@link InterruptedException}.
+     * @param job the job to run
+     * @param name the human-readable name of the job
+     */
+    public BackgroundJob newJob(Runnable job, String name);
+
+    /**
+     * {@link BackgroundJob#terminate() Terminates} all live
+     * {@link BackgroundJob background jobs} created by this factory, but does not wait for
+     * completion.
+     * @see BackgroundJob#terminate()
+     * @see #waitForTerminationOfAll(long)
+     */
+    public void terminateAll();
+
+    /**
+     * Checks whether all live {@link BackgroundJob background jobs} created by this factory are
+     * terminated.
+     * @return {@code true} if all jobs are terminated.
+     */
+    public boolean allTerminated();
+
+    /**
+     * Waits for all live {@link BackgroundJob background jobs} created by this factory to
+     * terminate. Note that this method makes a snapshot of the currently live jobs, so jobs
+     * created after the wait has started will not be waited upon. This method returns when all
```

Comment:
The intended purpose of this is that you can keep using the background job factory after having terminated all of its live jobs. Maybe this serves no real purpose and should be removed. 

Blocking threads on new job creation sounds like a bad idea, maybe it will be more useful to let `newJob` return `null` immediately on such a condition (and lose the ability to use the factory afterwards)?